applications/serial_lte_modem/src/slm_at_tcp_proxy.c

```diff
@@ -1115,3 +1122,29 @@ int slm_at_tcp_proxy_uninit(void)
 
 	return 0;
 }
+
+/**@brief API to get datamode from external
+ */
+bool slm_tcp_get_datamode(void)
+{
+	return proxy.datamode;
+}
+
+/**@brief API to set datamode off from external
+ */
+void slm_tcp_set_datamode_off(void)
+{
+	proxy.datamode = false;
+}
+
```

Comment:
How about slm_tcp_set_datamode(bool datamode) to set the datamode? It can also used by additional AT command to enter data mode from TCP/UDP server of it started as non-datamode.

Reply:
This was the same as my first commit, later I realize there is no Enable datamode requirement from at_host to proxy yet, only Disable so the API is changed to be like this as of now. Re-enter datamode only make sense for TCP/UDP clients. To support Re-Enter we need to change the mechanism of datamode configuration, as re-enter means it was in datamode before. Current mechanism will lose this info if MCU quit datamode once.